services/galley/src/Galley/Effects/BackendNotificationQueueAccess.hs

```diff
@@ -23,5 +22,49 @@ data BackendNotificationQueueAccess m a where
     f (Remote x) ->
     (Remote [x] -> FedQueueClient c a) ->
     BackendNotificationQueueAccess m (Either FederationError [Remote a])
+  EnqueueNotificationsConcurrentlyBuckets ::
+    (KnownComponent c, Foldable f, Functor f) =>
+    Q.DeliveryMode ->
+    f (Remote x) ->
+    (Remote x -> FedQueueClient c a) ->
+    BackendNotificationQueueAccess m (Either FederationError [Remote a])
+
+enqueueNotification ::
+  ( KnownComponent c,
+    Member (Error FederationError) r,
+    Member BackendNotificationQueueAccess r
+  ) =>
+  Q.DeliveryMode ->
+  Remote x ->
+  FedQueueClient c a ->
+  Sem r a
+enqueueNotification m r q = send (EnqueueNotification m r q) >>= either throw pure
+
+enqueueNotificationsConcurrently ::
+  ( KnownComponent c,
+    Foldable f,
+    Functor f,
+    Member (Error FederationError) r,
+    Member BackendNotificationQueueAccess r
+  ) =>
+  Q.DeliveryMode ->
+  f (Remote x) ->
+  (Remote [x] -> FedQueueClient c a) ->
+  Sem r [Remote a]
+enqueueNotificationsConcurrently m r q =
+  send (EnqueueNotificationsConcurrently m r q)
+    >>= either throw pure
 
-makeSem ''BackendNotificationQueueAccess
+enqueueNotificationsConcurrentlyBuckets ::
+  ( KnownComponent c,
+    Foldable f,
+    Functor f,
+    Member (Error FederationError) r,
+    Member BackendNotificationQueueAccess r
+  ) =>
+  Q.DeliveryMode ->
+  f (Remote x) ->
+  (Remote x -> FedQueueClient c a) ->
+  Sem r [Remote a]
+enqueueNotificationsConcurrentlyBuckets m r q =
+  send (EnqueueNotificationsConcurrentlyBuckets m r q) >>= either throw pure
```

Comment:
This looks absurd, the only exception that can happen is federation not being configured. Should we just remove these `Either`s from the API of this effect and just throw them in IO?

Reply:
Why absurd? The only error is not being able to talk to the queue. What's wrong with having those errors as `Either`?